### ISSUE TITLE
[Feature request] Option to block 60 fps, VP8/VP9, AV1

### ISSUE THREAD

**Issue by kaypoush** [FreeTubeApp/FreeTube#504]
I have a Chromebook which is not powerful enough to play 60 fps and/or VP8/VP9 videos without stuttering. An option to block them in FreeTube would be nice, something that an extension for browser like [h264ify](https://github.com/erkserkserks/h264ify) is able to do apparently. By blocking, I don't mean remove all HD options because normal HD videos are playing fine but more something like this: "720p" instead of "720p 60fps" option when choosing the quality, and "h264" instead of "VP8/VP9". It would be very nice for low end computers.

**Comment by PrestonN** [comment]
This is something that I will try to look into more for the rewrite that I'm working on.  Ideally it would be a simple toggle asking if you'd like to use 60fps when available.

In the mean time, if you update your settings to use the legacy player, you will be limited to 720p 30fps which should do what you want.  60fps will also never show up on anything lower than 720p so changing your default quality could help there as well.

**Comment by ylzhan** [comment]
Thank you! I tried legacy player but it's slower than dash mode I don't know why. My Chromebook (CB3-131) is a potato. Also 480p on YouTube is very bad quality the encoding is poorly done I think. What is weird is I have no issue to play HD videos even 1080p with mpv it's pretty fluid but the youtube on browser is so slow. I heard it's because I don't have hardware acceleration so the CPU is struggling. Anyway.

There is another extension but with more options

https://github.com/alextrv/enhanced-h264ify

Maybe you can look in the code to make your task easier. Thanks again.

**Comment by rawlife56** [comment]
Any update on this one ? @PrestonN

**Comment by GilgusMaximus** [comment]
For the moment no really that much progress here. I'm not too sure how easy this is because we are getting the information from a module and currently they give us 30fps still with vp9. This is mostly due to the fact that once YouTube activate vp9 for a video, they provide at least all resolutions of hd and up in vp9. Additionally we are a bit hindered by our video player's technology, which only supports mp4 formats and not webm formats

**Comment by PrestonN** [comment]
Since we now have a little more control over the DASH files that are being generated in the rewrite, it should be fairly easy to remove any 60fps options before the DASH file is created. It isn't a high priority at the moment but if someone wants to create a pull request for it then I'll be happy to take a look at it. Let me know if you need any assistance for getting it done.

**Comment by Albonycal** [comment]
Any updates? Vp9 is very hard on my CPU

**Comment by PrestonN** [comment]
Give the latest nightly builds a test. 60fps options have been broken out and it should be trivial to select the non-60fps variants of videos now. For this upcoming release, we probably won't have a toggle to prefer 30fps options however the framework is there now to allow it for a release after that.

**Comment by Ph0rk0z** [comment]
I don't care about 60FPS... needs to be H264 or my GPU can't decode it. Chrome on linux can do it, mpv with ytd can do it. If this can't do it, I can't use it as the cpu usage + temp will be high.

**Comment by ylzhan** [comment]
I just want to give a feedback on this as the author of this request. 60 fps or not is not important but it could be cool to have that option too if it's possible. My screen is 720p 60 hz so capable of 60 fps I don't remember why I focused on the 60 fps part. My gpu can decode MPEG2, H264 and VC1 according to vainfo. Anything else will be cpu decoded (so video lag if you have a potato chromebook like mine and not good for battery life anyway). 

Now I was using freetube with mpv (enabled hw and blocked av1, vp8 and vp9 in mpv.conf) and it was working fine but I didn't like to use two programs to play a video especially when there is a longer delay to start the video from mpv. Then I got va-api working in firefox but it's not correctly working since v98 (rdd sandbox bug). Then I searched for "freetube hardware acceleration" and found about this https://github.com/FreeTubeApp/FreeTube/issues/961 and saw that preston enabled accelerated video feature in electron (chromium) which freetube uses. Then I tested out this based on this https://chromium.googlesource.com/chromium/src/+/master/docs/gpu/vaapi.md and https://wiki.archlinux.org/title/Chromium#Hardware_video_acceleration :

`freetube --enable-features=VaapiVideoDecoder --use-gl=egl --disable-features=UseChromeOSDirectVideoDecoder --ignore-gpu-blocklist`

played any video up to 1080p (even in 60 fps to show you why I was wrong by saying that my chromebook had difficulties with 60 fps) (settings for freetube : dash default 1080p) and checked with intel-gpu-top it's correctly working (video is not 0% when playback = gpu decoded).

you can replace the appropriate part with this in your custom freetube.desktop file in ~/.local/share/applications (you can create it as a copy based on the original one located in /usr/share/applications, the custom desktop file will be prioritized):

`Exec=/opt/FreeTube/freetube %U --enable-features=VaapiVideoDecoder --use-gl=egl --disable-features=UseChromeOSDirectVideoDecoder --ignore-gpu-blocklist`

(By the way, you need VA-API driver installed and packet name depends on your gpu. Mine is libva-intel-driver and I also have this LIBVA_DRIVER_NAME=i965 in my /etc/environment file. Two links that can help for this part: https://wiki.archlinux.org/title/Hardware_video_acceleration#Installation and https://wiki.debian.org/HardwareVideoAcceleration#Installation

So basically anything works fine. The feature request is still useful IMO but not a priority for me at least. Thanks preston for the great work.

**Comment by merspieler** [comment]
isn't it possible to differenciate the 720p and 720p60 option in the default resolution setting?

**Comment by aryaritesh08** [comment]
> I have a Chromebook which is not powerful enough to play 60 fps and/or VP8/VP9 videos without stuttering. An option to block them in FreeTube would be nice, something that an extension for browser like [h264ify](https://github.com/erkserkserks/h264ify) is able to do apparently. By blocking, I don't mean remove all HD options because normal HD videos are playing fine but more something like this: "720p" instead of "720p 60fps" option when choosing the quality, and "h264" instead of "VP8/VP9". It would be very nice for low end computers.

Same bro. Same with my pc. At first freetube had this feature but after the latest update, it's not there. YouTube sucks.

**Comment by Pentaphon** [comment]
Can we do this now that shaka-player is being used? I'd like to block vp9 and av1 for my older laptop.

**Comment by artenax** [comment]
AV1 is slow in HD. I would like to block it.

**Comment by freakedenough** [comment]
i need these options as well. i want 1080p with h264, not vp9.

**Comment by artenax** [comment]
You might find it interesting. I replaced FreeTube with browser-based V3 https://github.com/vorapis-yt/v3/
It has more control and pretty good performance.

**Comment by solisinvictum** [comment]
I just had the same Problem. But with "newer" Hardware. Intel i7 6gen to Intel i7 9gen.

Sadly, Youtube pushing more and more AV1, these got nearly unuseable and obsolete. Because Video or Audio is stuttering like af. And that with a 4.2ghz overclocked 8 threads cpu (6700k for example).

(I use AV1 my self, to save up space on my media server, but in this case, jellyfin simple transcode anything to h264. but that another topic)

It took me a long time, to get Freetube working with hardware acceleration and ONLY h264, but here we go. max resolution is this way only 1080p:

freetube --ignore-gpu-blacklist --ignore-gpu-blocklist --enable-features=VaapiVideoDecoder --disable-features=UseChromeOSDirectVideoDecoder,AV1Decoder,UseAV1,VP9Decoder,UseVP9 --use-gl=egl

This tells Chromium/Electron to disable AV1 and VP9. After that, Freetube hasnt any choice left and uses h264/avc1.

Edit: because i use Freetube on my Media/TV-PCs, i need to add this "Patch" to 6 different machines. To lazy for that. comming weekend i create a AUR Package, which patchs the .desktop file.

**Comment by Heavygrass** [comment]
@solisinvictum

Hi, I'm getting lots of these errors with your options in Cinnamon:
`ERROR:gl_factory.cc(102)] Requested GL implementation (gl=egl-gles2,angle=none) not found in allowed implementations: [(gl=egl-angle,angle=default)]`

with the latest flatpak version on old Intel CPU (there's very old version in the distro repo).
Run it as
`/usr/bin/flatpak run --branch
=stable --arch=x86_64 --command=/app/bin/run.sh --file-forwarding io.freetubeapp.FreeTube --ignore-gpu-blacklist 
--ignore-gpu-blocklist --enable-features=VaapiVideoDecoder --disable-features=UseChromeOSDirectVideoDecoder,AV1Decoder,UseAV1,VP9Decoder,UseVP9 --use-gl=egl`

Do you have a link to the description of the options?
Whats `--use-gl=egl` option mean?

Already allowed D-Bus access in flatpak.

**Comment by Heavygrass** [comment]
> Can we do this now that shaka-player is being used?

Freetube no longer uses chromium?

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Due to the limitations we're working with, this won't be feasible for the foreseeable future.

**Comment by Pentaphon** [comment]
> Due to the limitations we're working with, this won't be feasible for the foreseeable future.

@efb4f5ff-1298-471a-8973-3d47447115dc I don't think it's fair to close the issue as not planned unless the team doesn't want to implement the feature. It may not be feasible for the foreseeable future, but what about the future past that? What about blocking future formats like AV2? What if a new solution for connecting to Youtube comes out?
